lib/order.rb

```diff
@@ -0,0 +1,66 @@
+require_relative 'customer'
+class Order
+  attr_reader :id, :customer 
+  attr_accessor :products, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status=:pending)
+    @id = id
+    @customer = customer 
+    @products = products 
+    if [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
+       @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError.new("Bogus status")
+    end 
+  end 
+
+  def total
+   if @products.length == 0 
+      return 0.0
+   end 
+   total = 0
+   @products.each do |product, price|
+    total += price
+   end 
+   return ('%.2f' %(total * 1.075)).to_f
+  end 
+
+  def add_product(name, price)
+    if @products.include?(name)
+      raise ArgumentError.new("This product already present")
+    else
+      @products["#{name}"] = price 
+    end 
+  end 
+
+  def remove_product(name)
+    if @products.include?(name)
+      @products.delete(name)
+    else 
+      raise ArgumentError.new("did not find that product")
+    end 
+  end
+
+  def self.all
+   array_of_orders = []
+    CSV.foreach("../data/orders.csv") do |row|
```

Comment:
In general, paths should be relative to the home directory as that's where the code will be run (for instance, with rake).